OSDOCS-19969: Gateway API backendTLSPolicy for securing HTTPRoutes (#112677)
openshift/openshift-docs: modules/httproute-tls-options.adoc

// Module included in the following assemblies:
//
// * networking/ingress_load_balancing/configuring_gateway_api/securing-httproutes.adoc

:_mod-docs-content-type: CONCEPT
[id="httproute-tls-options_{context}"]

> = HTTPRoute Transport Layer Security (TLS) configuration

Reviewer: ```suggestion
= HTTPRoute TLS configuration options
```

> = Securing application traffic using HTTPRoute TLS modes

Reviewer 2: The TLS modes are on the listeners.  I would call this section
```suggestion
= Understanding HTTPRoute Transport Layer Security (TLS) configuration
```

Author: Thanks! I retitled this section "HTTPRoute Transport Layer Security (TLS) configuration". Your suggested change helped me notice that I had also structured this title incorrectly for a concept model (that's why I removed the gerund)

[role="_abstract"]

> To protect sensitive data and meet security compliance requirements, secure your application traffic by applying edge or re-encrypt TLS modes to your HTTPRoutes. 

Reviewer 2: ```suggestion
To protect sensitive data and meet security compliance requirements, secure your application traffic by applying edge or re-encrypt TLS modes to the gateway listeners that are used by your HTTPRoutes. 
```

Author: Got it, thanks!

> Depending on your specific goals, you can choose between edge and re-encrypt termination. Both of these modes also exist on traditional {product-title} routes.

Reviewer: ```suggestion
```
repetitive and not needed
there is a lot of AI-generated verbosity in this PR; please apply minimalism when reviewing

> Depending on your security architecture and performance needs, choose the TLS termination mode that best aligns with your goals. Both of these modes also exist on legacy {product-title} routes.

Reviewer 2: Legacy is not the best term.  We will keep these around.  You could say:

```suggestion
Depending on your security architecture and performance needs, choose the TLS termination mode that best aligns with your goals. Both of these modes also exist on traditional {product-title} routes.
```

Author: Excellent, thanks

> Depending on your security architecture and performance needs, choose the TLS termination mode that best aligns with your goals. Both of these modes also exist on traditional {product-title} routes.

Reviewer 2: The first line is repetitive when seen in combination with the previous module in netlify.
Also, can we please add a link to the definitions for OpenShift Routes?
```suggestion
HTTPRoute supports two TLS termination modes: edge and re-encrypt.   Depending on your security architecture and performance needs, choose the TLS termination mode that best aligns with your goals. Both of these modes also exist on traditional {product-title} routes, as described in Securing Routes <link needed>.
```

Author: Great catch! I've rewritten the concept module's abstract to remove the redundancy. The new abstract focuses on user goals to comply with our JTBD strategy and DITA short description guidelines. 

About the added link, I completely agree this context needs to be included. However, we're no longer allowed to add links and cross references to concept modules. (It's a DITA thing.) So, I've added the cross-reference to the Additional Resources section in the main assembly.

Edge termination:: 

> To secure traffic from the client to the gateway without adding the overhead of internal encryption, use edge TLS termination. This mode encrypts the client traffic, but allows unencrypted traffic to pass from the gateway to your internal services. This configuration is ideal for scenarios where you must encrypt user logins, but want to trade internal traffic encryption for faster application performance.

Reviewer: ```suggestion
To secure traffic from the client to the gateway without internal encryption, use edge TLS termination. This configuration is ideal for scenarios where you must encrypt user logins, but want to trade internal traffic encryption for faster application performance. This mode encrypts the client traffic, but allows unencrypted traffic to pass from the gateway to your internal services. 
```

> To secure external connections without adding the overhead of backend encryption, use edge TLS termination. This mode encrypts traffic from the client to the gateway, but allows clear-text communication from the gateway to your internal services. This configuration is ideal for scenarios where you must secure user logins at the perimeter, but want to trade internal traffic encryption for faster application performance.

Reviewer 2: The terms external, backend, perimeter, communication, are a little confusing here.
```suggestion
In cases where you need to secure traffic  from the client to the gateway, but not traffic from the gateway to the internal service, use edge TLS termination. This mode encrypts traffic from the client to the gateway, but allows unencrypted traffic from the gateway to your internal services. This configuration is ideal for scenarios where you must encrypt user logins, but want to trade internal traffic encryption for faster application performance.
```

Author: Thanks, I updated this section

Re-encrypt termination:: 

> To meet strict security requirements while still allowing the gateway to inspect payloads, use re-encrypt TLS termination. This mode encrypts traffic from the client to the gateway (edge termination), and then re-encrypts it from the gateway to the destination service (backend termination). This allows you to efficiently route traffic or block unauthorized access at the gateway without eliminating backend traffic encryption.

Reviewer: ```suggestion
To meet strict security requirements while still allowing the gateway to inspect payloads, use re-encrypt TLS termination. You can efficiently route traffic or block unauthorized access at the gateway without eliminating backend traffic encryption. This mode enforces edge and backend termination. It encrypts traffic from the client to the gateway, and then re-encrypts it from the gateway to the destination service. 
```

> [NOTE]

Reviewer: ```suggestion
[IMPORTANT]
```

> To meet strict security requirements while still allowing the gateway to inspect payloads, use re-encrypt TLS termination. This mode encrypts traffic from the client to the gateway (edge termination), and then re-encrypts it from the gateway to the destination service (backend termination). This allows you to efficiently route traffic or block unauthorized access at the gateway without compromising your backend encryption.

Reviewer 2: ```suggestion
To meet strict security requirements while still allowing the gateway to inspect payloads, use re-encrypt TLS termination. This mode encrypts traffic from the client to the gateway (edge termination), and then re-encrypts it from the gateway to the destination service (backend termination). This allows you to efficiently route traffic or block unauthorized access at the gateway without eliminating backend traffic encryption.
```

Author: Thanks, I made this very needed update

====

> {product-title} does not currently support passthrough termination for Gateway API. Passthrough termination provides direct traffic encryption from the client to the service, and is only supported in traditional {product-title} routes.

Reviewer: ```suggestion
{product-title} does not support passthrough termination for Gateway API. Passthrough termination provides direct traffic encryption from the client to the service, and is only supported in traditional {product-title} routes.
```

> If you require end-to-end encryption without gateway termination, {product-title} routes support a third mode called passthrough termination. However, {product-title} does not currently support passthrough termination for Gateway API.

Reviewer 2: ```suggestion
{product-title} does not currently support passthrough termination for Gateway API.  Passthrough termination is traffic encryption from client to service, and is only supported in {product-title} routes.
```

Author: Excellent! Thanks!

> // TODO: In a future release, passthrough termination will be enabled for a new type of Gateway API route called TLSRoute. Update this note once the TLSRoute feature officially GAs.

Reviewer: ```suggestion
```
